extensions/storage/statedb/statedb.go

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package statedb
+
+import (
+	gossipapi "github.com/hyperledger/fabric/extensions/gossip/api"
+)
+
+//AddCCUpgradeHandler adds chaincode upgrade handler to blockpublisher
+func AddCCUpgradeHandler(chainName string, handler gossipapi.ChaincodeUpgradeHandler) {
```

Comment:
I don't see the point to this function. The caller can just call AddCCUpgradeHandler on the BlockPublisher directly. In fabric-mod this function is a no-op anyway.

Reply:
Initially I planned like that only, but later realized, lscc cache update gonna occur twice in case of committer. So this handler is endorser only.

Reply:
OK